app/web/features/communities/events/LongEventCard.tsx

```diff
@@ -1,145 +1,164 @@
-    alignItems: "center",
+  cardContent: {
+    width: "75%",
     display: "flex",
-    marginBlockStart: theme.spacing(1),
+    flexDirection: "column",
+    justifyContent: "space-between",
   },
-  attendeesCountContainer: {
-    alignItems: "center",
+  eventInfo: {
     display: "flex",
+    justifyContent: "flex-end",
+    flexDirection: "column",
   },
-  icon: {
-    display: "block",
-    fontSize: "1.25rem",
-    lineHeight: 1.5,
-    marginInlineEnd: theme.spacing(0.5),
+  row: {
+    display: "flex",
+    justifyContent: "space-between",
   },
-  image: {
-    objectFit: ({ eventImageSrc }) =>
-      eventImageSrc === eventImagePlaceholderUrl ? "contain" : "cover",
-    height: 80,
-    [theme.breakpoints.up("md")]: {
-      height: 120,
+  tags: {
+    minWidth: theme.spacing(15),
+    [theme.breakpoints.down("xs")]: {
+      minWidth: theme.spacing(10),
     },
-    width: "100%",
-    gridArea: "image",
   },
-  onlineOrOfflineInfo: {
-    fontWeight: "bold",
-    color: theme.palette.grey[600],
-  },
-  content: {
-    gridArea: "content",
+  title: {
+    display: "-webkit-box",
+    lineClamp: 2,
+    boxOrient: "vertical",
+    overflow: "hidden",
+    textOverflow: "ellipsis",
+    maxHeight: "3em" /* Approximate height for 2 lines of text */,
+    lineHeight: "1.5em",
+    paddingRight: theme.spacing(2),
   },
 }));
 
-export interface LongEventCardProps {
+const LongEventCard = ({
+  event,
+  userId,
+}: {
   event: Event.AsObject;
-}
-
-export default function LongEventCard({ event }: LongEventCardProps) {
-  const { t } = useTranslation([COMMUNITIES]);
+  userId: number | null | undefined;
+}) => {
   const classes = useStyles({
     eventImageSrc: event.photoUrl || eventImagePlaceholderUrl,
   });
-  const theme = useTheme();
-  const isBelowLg = useMediaQuery(theme.breakpoints.down("md"));
+  const { t } = useTranslation([COMMUNITIES]);
 
-  const truncatedContent = useMemo(
-    () =>
-      getContentSummary({
-        originalContent: event.content,
-        maxLength: isBelowLg ? 120 : 300,
-      }),
-    [event.content, isBelowLg]
-  );
-  const startTime = dayjs(timestamp2Date(event.startTime!));
+  const startTime = dayjs(timestamp2Date(event.startTime!)).format("llll");
+
+  const renderTags = () => {
+    const isCreatedByMe = event.creatorUserId === userId;
+    const isOnline = event.onlineInformation?.link !== undefined;
+    const isCancelled = event.isCancelled;
+
+    return (
+      <div className={classes.tags}>
+        {isCreatedByMe && (
+          <Pill variant="rounded">{t("communities:created_by_me")}</Pill>
+        )}
+        {isOnline && <Pill variant="rounded">{t("communities:online")}</Pill>}
+        {isCancelled && (
+          <Pill
+            backgroundColor={theme.palette.error.main}
+            color={theme.palette.common.white}
+            variant="rounded"
+          >
+            {t("communities:cancelled")}
+          </Pill>
+        )}
+      </div>
+    );
+  };
```

Comment:
A minor thing (and it's easier to comment here than on the final giant PR) since it's pure UI and so it doesn't change much - but is there any reason to not render this inline with the JSX below?

And if it's done for colocating the computed variables closer to the JSX, why not break it up into a new component? After all, the body of a component is more or less the `render` function for JSX.

Reply:
Ahhh hmm I think this is just a style thing I picked up at my last company to break things into render functions. I can certainly put it into the body or break it into a new component. I'll do that in the main feature branch!